includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -663,11 +663,14 @@ private function get_request_payment_data( \WP_REST_Request $request ) {
 
 	/**
 	 * Proceed with current request using new login session (to ensure consistent nonce).
+	 * Only apply during the checkout process with the account creation.
 	 *
 	 * @param string $cookie New cookie value.
 	 */
 	public function set_cookie_on_current_request( $cookie ) {
-		$_COOKIE[ LOGGED_IN_COOKIE ] = $cookie;
+		if ( defined( 'WOOCOMMERCE_CHECKOUT' ) && WOOCOMMERCE_CHECKOUT && did_action( 'woocommerce_created_customer' ) > 0 ) {
```

Comment:
>did_action( 'woocommerce_created_customer' )

Does that mean it will still happen for users who will create a new account?

Reply:
No, this `did_action` only matters when shoppers submit the checkout request with the option of creating a new account. The flow having issue, as mentioned in #7992, is with the log-in flow for an existing user.

Reply:
>The flow having issue, as mentioned in https://github.com/Automattic/woocommerce-payments/issues/7992, is with the log-in flow for an existing user.

Can we double check, if that's the case? And log a separate issue for that flow?

Reply:
@naman03malhotra there is no need for a separate issue, this PR handles both flows.

Reply:
I mean, does it also happen when a shopper creates a new account while checkout? So I was talking about that flow.

Reply:
> During the checkout registration, the cart contents are already moved into the order, so losing them is not an issue.

Thanks @RadoslavGeorgiev for this answer.